src/components/user.js

```diff
@@ -98,35 +108,35 @@ export default class User extends React.Component {
             currentUser[obj][name] = value;
             formErrors[name]       = this._validateField(name, value);
             if (name === FIELD_NAMES.name) {
-                showUserError = false;
-                addDuplicate = false;
+                forceErrorDuplicate = false;
             }
-            this.setState({ currentUser, formErrors, showUserError, addDuplicate });
+            this.setState({ currentUser, formErrors, forceErrorDuplicate });
         }
         e.preventDefault();
     };
 
     _handleClickSubmit = e => {
-        const { currentUser, addDuplicate, formErrors } = this.state;
-        const newUser                                   = new UserInfo(currentUser.info, currentUser.address, currentUser.company);
-        const resultValid                               = this._validateFields(currentUser.info);
-        const isUserInfoValid                           = !resultValid[FIELD_NAMES.name] &&
-                                                          !resultValid[FIELD_NAMES.phone] &&
-                                                          !resultValid[FIELD_NAMES.email];
-
-        if (isUserInfoValid) {
-            UserAction.addNewUser(newUser, addDuplicate);
-            this.setState({ formErrors, addDuplicate: !addDuplicate, showUserError: true });
+        const { currentUser, forceErrorDuplicate, formErrors } = this.state;
+        const newUser                                          = new UserInfo(currentUser.info, currentUser.address, currentUser.company);
+        const resultValid                                      = this._validateFields(currentUser.info);
+
+        if (this._isUserInfoValid(resultValid)) {
+            UserAction.addNewUser(newUser, forceErrorDuplicate);
```

Comment:
(proposal)
Now it's better. But I can offer you to rename `this.state.forceErrorDuplicate` -> `this.state.hasDuplicateError`.